javav2/example_code/dynamodb/src/main/java/com/example/enhanced/dynamodb/BatchWriteItems.java

```diff
@@ -0,0 +1,179 @@
+//snippet-sourcedescription:[BatchWriteItems.java demonstrates how to insert many items into an AWS DynamoDB table by using the enhanced client]
+//snippet-keyword:[SDK for Java 2.0]
+//snippet-keyword:[Code Sample]
+//snippet-service:[dynamodb]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[3/15/2020]
+//snippet-sourceauthor:[scmacdon-aws]
+
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   This file is licensed under the Apache License, Version 2.0 (the "License").
+   You may not use this file except in compliance with the License. A copy of
+   the License is located at
+    http://aws.amazon.com/apache2.0/
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+package com.example.enhanced.dynamodb;
+
+// snippet-start:[dynamodb.java2.enhanced.batchitems.import]
+import static software.amazon.awssdk.enhanced.dynamodb.mapper.StaticAttributeTags.primaryPartitionKey;
+import static software.amazon.awssdk.enhanced.dynamodb.mapper.StaticAttributeTags.primarySortKey;
+import static software.amazon.awssdk.enhanced.dynamodb.mapper.StaticAttributeTags.secondaryPartitionKey;
+import static software.amazon.awssdk.enhanced.dynamodb.mapper.StaticAttributeTags.secondarySortKey;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbTable;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.StaticTableSchema;
+import software.amazon.awssdk.enhanced.dynamodb.model.BatchWriteItemEnhancedRequest;
+import software.amazon.awssdk.enhanced.dynamodb.model.WriteBatch;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.DynamoDbException;
+import software.amazon.awssdk.services.dynamodb.model.ProvisionedThroughput;
+// snippet-end:[dynamodb.java2.enhanced.batchitems.import]
+
+public class BatchWriteItems {
+
+    private static final ProvisionedThroughput DEFAULT_PROVISIONED_THROUGHPUT =
+            ProvisionedThroughput.builder()
+                    .readCapacityUnits(50L)
+                    .writeCapacityUnits(50L)
+                    .build();
+
+    private static final TableSchema<Record> TABLE_SCHEMA =
+            StaticTableSchema.builder(Record.class)
+                    .newItemSupplier(Record::new)
```

Comment:
Style: Personally I prefer to use descriptive examples like Customer instead of generic ones like Record (same with the attributes and the index names). I feel like it's easier for people to 'get' what we're trying to illustrate or can see how it's used in context. What do you think?

Reply:
We have another example in an article that will help drive this point.  In that article, we use this: 

public class **GreetingItems** {

    //Set up Data Members that correspond to columns in the Work table
    private String id;
    private String name;
    private String message;
    private String title;
...

Between these two examples, our customers will understand this mapping.